serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/auth/OidcRemoteTokenServices.java

```diff
@@ -0,0 +1,121 @@
+        restTemplate = new RestTemplate();
+        ((RestTemplate) restTemplate).setErrorHandler(new DefaultResponseErrorHandler() {
+            @Override
+            // Ignore 400
+            public void handleError(ClientHttpResponse response) throws IOException {
+                if (response.getRawStatusCode() != 400) {
+                    super.handleError(response);
+                }
+            }
+        });
+
+        setCheckTokenEndpointUrl(props.getTokenInfoUri());
+        setClientId(props.getClientId());
+        setClientSecret(props.getClientSecret());
+
+    }
+
+    @Override
+    public OAuth2Authentication loadAuthentication(String accessToken) throws AuthenticationException, InvalidTokenException {
+        MultiValueMap<String, String> formData = new LinkedMultiValueMap<String, String>();
+        formData.add(tokenName, accessToken);
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", getAuthorizationHeader(clientId, clientSecret));
+        Map<String, Object> map = postForMap(checkTokenEndpointUrl, formData, headers);
+
+        if (map.containsKey("error")) {
+            log.warn("check_token returned error: " + map.get("error"));
+            throw new InvalidTokenException(accessToken);
+        }
+
+        if (map.containsKey("active") && !(Boolean)map.get("active")) {
+            log.debug("check_token returned active -> false for token "+accessToken);
+            throw new InvalidTokenException(accessToken);
+        }
+
+        return tokenConverter.extractAuthentication(map);
+    }
+
+    @Override
+    public OAuth2AccessToken readAccessToken(String accessToken) {
+        throw new UnsupportedOperationException("Not supported: read access token");
+    }
+
+    private String getAuthorizationHeader(String clientId, String clientSecret) {
+
+        if(clientId == null || clientSecret == null) {
+            log.warn("Null Client ID or Client Secret detected. Endpoint that requires authentication will reject request with 401 error.");
+        }
+
+        String creds = String.format("%s:%s", clientId, clientSecret);
+        try {
+            return "Basic " + new String(Base64.encode(creds.getBytes("UTF-8")));
+        }
+        catch (UnsupportedEncodingException e) {
+            throw new IllegalStateException("Could not convert String");
+        }
+    }
+
+    private Map<String, Object> postForMap(String path, MultiValueMap<String, String> formData, HttpHeaders headers) {
+        if (headers.getContentType() == null) {
+            headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        }
+        @SuppressWarnings("rawtypes")
+        Map map = restTemplate.exchange(path, HttpMethod.POST,
+                new HttpEntity<MultiValueMap<String, String>>(formData, headers), Map.class).getBody();
+        @SuppressWarnings("unchecked")
+        Map<String, Object> result = map;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1488)